TestFoundation/TestNSStream.swift

```diff
@@ -124,6 +132,107 @@ class TestNSStream : XCTestCase {
         XCTAssertEqual(Stream.Status.error, fileStream.streamStatus)
     }
     
+    /* This test is the success case for Streams that are in the blessed list with the
+     * dataGetBuffer as part of the call back e.g
+     * static const struct _CFStreamCallBacksV1 readDataCallBacks = {1, readDataCreate,readDataFinalize, readDataCopyDescription, readDataOpen, NULL, dataRead, dataGetBuffer, dataCanRead, NULL, NULL, NULL, NULL, NULL, NULL, readDataSchedule, NULL};
+     */
+    func test_InputStreamGetBufferSuccessFromBlessedList(){
+        let message: NSString = "Hello, playground"
+        let messageData: Data = message.data(using: String.Encoding.utf8.rawValue)!
+        let dataStream = InputStream(data: messageData)
+        XCTAssertEqual(Stream.Status.notOpen, dataStream.streamStatus)
+        dataStream.open()
+        XCTAssertEqual(Stream.Status.open, dataStream.streamStatus)
+        let buffer = UnsafeMutablePointer<UnsafeMutablePointer<UInt8>?>.allocate(capacity: 1)
```

Comment:
I think both suggestions are good. The stream doesn't know what type is held in the memory, so it should take UnsafeMutableRawPointer. UnsafeBuffer is a better way to pass pointer+ length. You can't combine the two yet because I haven't gotten around to proposing UnsafeRawBufferPointer, but I expect it to be an obvious addition.